### PR TITLE
Implement rest of the planned endpoints for the new API

### LINKED ISSUE #9: Include global margin into the database.

### LINKED ISSUE #12: Implement admin part of new api.

### PR DESCRIPTION
**Pull request by dogamak**
**Stuff to do:**
- [x] Use the existing methods in `historyService` instead of the ones I made before realizing `historyService` existed.
- [x] Remove redundant endpoints (e.g. `/api/v1/admin/user/:userId/purchaseHistory/:purchaseId` and `/api/v1/admin/product/:barcode/purchaseHistory/:purchaseId`)
- [x] Endpoints for querying and modifying preferences.
    - Should `/api/v1/admin/defaultMargin` be removed in favour of `/api/v1/admin/preferences/globalDefaultMargin`?
- [x] More logging.
- [x] Deletion flags, instead of hard deletion, for objects other than products also? Boxes and categories don't seem to have any foreign key constraints in the way and they aren't referenced from the history tables, so just deleting them should be fine.

### CODE REVIEW

**Comment by dogamak** [comment]
Addresses also the following issues:
- #9: This PR includes a mechanism for storing global preferences (including the global default profit margin) in the database.
- #36: This PR introduces a `deleted` flag for the `RVITEM` table, which denotes that the product has been deleted.
- #12: Implements all remaining administration API endpoints.

**Comment by Tuupertunut** [comment]
Do buy-ins require their own functions in the box store and product store? I think they can be implemented with only productStore.updateProduct().

File: src/db/migrations/20201010200201_preferences.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+exports.up = async (knex) => {
+    if (!(await knex.schema.hasTable('PREFERENCES'))) {
+        await knex.schema.createTable('PREFERENCES', (table) => {
```
Comment (Tuupertunut) [inline src/db/migrations/20201010200201_preferences.js:3]: This table lacks primary key, is this a problem?

File: src/db/preferences.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    validate: (value) =>
+        typeof value !== 'number' ? ['value should be a number'] :
+            !Number.isInteger(value) ? ['value should be an integer'] : []
+};
+
+module.exports.preferences = {
+    GLOBAL_DEFAULT_MARGIN: {
+        ... floatPreference,
+        key: 'globalDefaultMargin',
+        default: 0.05
```
Comment (Tuupertunut) [inline src/db/preferences.js:86]: Should these default values be initialized in the database instead?

**Pull request merged**